templates/common/security-groups/DSMSGIngressRules.template

```diff
@@ -0,0 +1,179 @@
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMHeartbeatIngressFromELB:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMRelayIngressFromELB:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMAgentIngress:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4118'
+      ToPort: '4118'
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMConsoleIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMHeartbeatIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMRelayIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMConsoleIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      CidrIp: 192.168.0.0/16
+  DSMHeartbeatIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      CidrIp: 192.168.0.0/16
+  DSMRelayIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      CidrIp: 192.168.0.0/16
+  DSMConsoleIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      CidrIp: 10.0.0.0/8
+  DSMHeartbeatIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      CidrIp: 10.0.0.0/8
+  DSMRelayIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      CidrIp: 10.0.0.0/8
```

Comment:
I see. Would be something then to consider based on customer feedback on flexibility. Good for now though.